### PR TITLE
DIRECTOR: Fixes for The Dark Eye

### PR DESCRIPTION
**Pull request by moralrecordings**

### CODE REVIEW

File: engines/director/castmember/filmloop.cpp [hunk 1]
```diff
@@ -365,19 +365,24 @@ void FilmLoopCastMember::load() {
...
+		}
+
+		if (loop) {
+			debugC(2, kDebugLoading, "****** Loading '%s' id: %d, %d bytes", tag2str(tag), filmLoopId, (int)loop->size());
+			loadFilmLoopDataD4(*loop);
+			delete loop;
 		} else {
-			warning("FilmLoopCastMember::load(): Expected 1 child for film loop cast, got %d", _children.size());
+			warning("FilmLoopCastMember::load(): No SCVW resource found in %d children", _children.size());
 		}
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/castmember/filmloop.cpp:386]: The film loop ID isn't known at this point. It's blank.

File: engines/director/castmember/bitmap.cpp [hunk 2]
```diff
@@ -534,11 +534,14 @@ void BitmapCastMember::load() {
...
+		for (auto &it : _children) {
+			if (it.tag == MKTAG('B', 'I', 'T', 'D')) {
+				imgId = it.index;
+				tag = it.tag;
 
-			pic = _cast->getResource(tag, imgId);
+				pic = _cast->getResource(tag, imgId);
+				break;
+			}
 		}
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/castmember/bitmap.cpp:545]: On average there is one element in _children. It's fine.

File: engines/director/castmember/text.cpp [hunk 3]
```diff
@@ -413,11 +413,20 @@ void TextCastMember::load() {
...
+	uint stxtid = 0;
+	if (_cast->_version >= kFileVer400) {
+		for (auto &it : _children) {
+			if (it.tag == MKTAG('S', 'T', 'X', 'T')) {
+				stxtid = it.index;
+				break;
+			}
+		}
+		if (!stxtid) {
+			warning("TextCastMember::load(): No STXT resource found in %d children", _children.size());
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/castmember/text.cpp:425]: Not finding a resource will mean that the movie playback will be broken. We want users to be able to see it.

File: engines/director/castmember/filmloop.cpp [hunk 4]
```diff
@@ -370,19 +370,24 @@ void FilmLoopCastMember::load() {
...
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/castmember/filmloop.cpp:391]: This makes the error more wordy for no reason.

File: engines/director/lingo/lingo-code.cpp [hunk 5]
```diff
@@ -1531,6 +1531,25 @@ void LC::call(const Common::String &name, int nargs, bool allowRetVal) {
...
+		if (funcSym.type != VOIDSYM) {
+			if (nargs == 0) {
+				g_lingo->_stack.push_back(Datum());
+				nargs = 1;
+			}
+			g_lingo->_stack[g_lingo->_stack.size() - nargs] = funcSym.target; // Set first arg to target
+			call(funcSym, nargs, allowRetVal);
+			return;
+		}
+	}
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/lingo/lingo-code.cpp:1551]: Is there even a suggestion in here?

File: engines/director/lingo/lingo.cpp [hunk 6]
```diff
@@ -1653,11 +1653,7 @@ Datum Lingo::varFetch(const Datum &var, bool silent) {
...
 			g_debugger->varReadHook(name);
-			// "me" is a special case; even if there is a local variable or argument called "me",
-			// the current me object will take precedence.
-			if (name == "me" && _state->me.type == OBJECT) {
-				return _state->me;
-			} else if (_state->localVars && _state->localVars->contains(name)) {
+			if (_state->localVars && _state->localVars->contains(name)) {
 				return (*_state->localVars)[name];
 			}
 			debugC(1, kDebugLingoExec, "varFetch: local variable %s not defined", name.c_str());
```
[the remark this answers is not in the document]
Comment (moralrecordings) [reply engines/director/lingo/lingo.cpp:1659]: `Datum result;` is fully initialised, you might want to consider learning C++

**Comment by sev-** [comment]
Thank you! Ignore the bots.

**Pull request merged**